js/interface.js

```diff
@@ -74,6 +83,25 @@ String.prototype.toCamelCase = function () {
   }).replace(/([^A-Z-a-z])/g, '').toLowerCase();
 };
 
+function waitForSocketConnection() {
+  if (socketConnected) {
```

Comment:
I would have treated `socketConnected` and `hasLoggedIn` separately as pointed out in my PR